protostar/testing/cairo_cheatcodes/expect_call_cairo_cheatcode.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+from typing import Callable, Optional
+
+from protostar.starknet import CheatcodeException
+from protostar.starknet import RawAddress, Address
+from protostar.starknet.cheaters.contracts import ContractsCheaterException
+from protostar.starknet.data_transformer import CairoOrPythonData
+from protostar.testing.cairo_cheatcodes.cairo_cheatcode import CairoCheatcode
+
+
+class ExpectCallCairoCheatcode(CairoCheatcode):
+    @property
+    def name(self) -> str:
+        return "expect_call"
+
+    def build(
+        self,
+    ) -> Callable[[RawAddress, str, Optional[CairoOrPythonData]], Callable]:
+        return self.expect_call
+
+    def expect_call(
+        self,
+        contract_address: RawAddress,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> Callable:
+        return asyncio.run(
+            self._expect_call(
+                contract_address=Address.from_user_input(contract_address),
+                function_name=function_name,
+                calldata=calldata,
+            )
+        )
+
+    async def _expect_call(
+        self,
+        contract_address: Address,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> Callable:
+        try:
+            return await self.cheaters.expects.expect_call(
+                contract_address=contract_address,
+                function_name=function_name,
+                calldata=calldata,
+            )
+        except ContractsCheaterException as exc:
```

Comment:
Is it a correct expection to catch?

Reply:
Hmm, right, we should catch `CheaterException`, thanks!

Reply:
[fix error type](https://github.com/software-mansion/protostar/pull/1363/commits/e95b228dec4bb0e767ae21e28e9712961a7cd2ba)